app/routes/core.rb

```diff
@@ -41,6 +41,20 @@ class Core < Sinatra::Application
         cache_control :private
       end
 
+      before do
+        if session[:access_token] && current_user.guest?
```

Comment:
Can we turn this into a guard clause to exit as early as possible? That way the rest of the method can be out-dented one level.